### PR TITLE
Patched multiple procs writing to same file

### PR DESCRIPTION
**Pull request by pbehne**
Console objects have a parameter 'output_file', that when true, writes console output to a file. Parallel simulations would write information from all processors to the same file, which can lead to bad behavior. This commit introduces a patch that only writes console content from the head proc to the file. In the case that --keep-cout is passed to the executable, each proc writes its console content to a separate file. The same bahavior can be achieved without setting output_file to true by instead passing '--keep-cout --redirect-stdout' to the executable.

Closes Ref #26174.

### CODE REVIEW

**Comment by lindsayad** [comment]
I would rebase this on next so that you get more targets passing (with all those solid mechanics changes)

**Comment by pbehne** [comment]
Hmm, I rebased to upstream/next and force pushed, but for some reason, that's not showing up on this PR. Any idea why?

File: test/tests/outputs/console/tests [hunk 1]
```diff
@@ -53,6 +53,33 @@
...
+  [./file_system_information_2procs_head]
+    # Test that file contains regex
+    type = CheckFiles
+    input = 'console.i'
+    cli_args = 'Outputs/screen/output_file=true Outputs/screen/file_base=console_file_system_information_out'
+    check_files = 'console_file_system_information_out.txt'
+    check_not_exists = 'console_file_system_information_out_0.txt console_file_system_information_out_1.txt'
+    recover = false
+    min_parallel = 2
+    max_parallel = 2
```
Comment (GiudGiud) [inline test/tests/outputs/console/tests:65]:
```suggestion
```

File: test/tests/outputs/console/tests [hunk 2]
```diff
@@ -53,6 +53,33 @@
...
+  [../]
+  [./file_system_information_2procs_all]
+    # Test that file contains regex
+    type = CheckFiles
+    input = 'console.i'
+    cli_args = 'Outputs/screen/output_file=true Outputs/screen/file_base=console_file_system_information_out --keep-cout'
+    check_files = 'console_file_system_information_out_0.txt console_file_system_information_out_1.txt'
+    recover = false
+    min_parallel = 2
+    max_parallel = 2
```
Comment (GiudGiud) [inline test/tests/outputs/console/tests:78]:
```suggestion
```

File: test/tests/outputs/console/tests [hunk 3]
```diff
@@ -53,6 +53,33 @@
...
+
+    requirement = "The system shall support outputting all processors' console information to separate files when run in parallel with --keep-cout."
```
Comment (GiudGiud) [inline test/tests/outputs/console/tests:80]:
```suggestion
    requirement = "The system shall support outputting all processors' console information to separate files when run in parallel with the --keep-cout command line option."
```

**Review by GiudGiud — CHANGES_REQUESTED** [review]
small changes. This is going to be convenient thanks for the fix

**Review by GiudGiud — APPROVED** [review]

**Comment by GiudGiud** [comment]
Thanks for the fix!

**Pull request merged**